tests/unit/test_common.cpp

```diff
@@ -477,10 +477,10 @@ TEST_F(SerializerTest, HNSWSerialzer) {
                              "Cannot load index: deprecated encoding version: 0");
 
     output.seekp(0, std::ios_base::beg);
-    Serializer::writeBinaryPOD(output, 42);
+    Serializer::writeBinaryPOD(output, 10042);
     output.flush();
     ASSERT_EXCEPTION_MESSAGE(HNSWFactory::NewIndex(this->file_name), std::runtime_error,
-                             "Cannot load index: bad encoding version: 42");
+                             "Cannot load index: bad encoding version: 10042");
```

Comment:
Those are still invalid for hnsw.
Consider updating the validation to be stricter.
Also, do we need such tests for  svs as well?

Reply:
The way bad encoding version is checked is by comparing the version from the binary file to the EncodingVersion_INVALID enum value (if the less than -> it's good), because we added EncodingVersion_SVS_V0 = 1000, the value 42 is no longer invalid.
I choose 10042 so the test will not have to change no matter the version increase.
This test can be relevant to SVS yes, I'll add one.

Reply:
👌